f8a_worker/storages/s3.py

```diff
@@ -195,6 +195,16 @@ def retrieve_dict(self, object_key):
         """ Retrieve a dictionary stored as JSON from S3 """
         return json.loads(self.retrieve_blob(object_key).decode())
 
+    def retrieve_key_list(self, prefix=None):
+        """ Retrieve all the keys in S3 based on a prefix"""
```

Comment:
This is needed in case we want to do force_graph_syncs. We can just do `selective_tasks = []` to sync all the data. I don't know, but I am assuming, we can speed up things with this approach since we can directly invoke GraphImporter task and still do the syncs directly without invoking any other un-necessary task. Does it make sense?

Reply:
> This is needed in case we want to do force_graph_syncs. We can just do selective_tasks = [] to sync all the data. I don't know, but I am assuming, we can speed up things with this approach since we can directly invoke GraphImporter task and still do the syncs directly without invoking any other un-necessary task. Does it make sense?

This is handled by our orchestrator - Selinon/Dispatcher that computes which tasks need to be run and if there are any that need to be run. In case you want to get all tasks for the given package, please create such method in S3 adapter. In this case there should be 2 methods - one in `S3Data` to handle package-version level data and another in `S3PackageData` to handle package-level data. `S3Data`'s method should accept ecosystem, package, version parameters and `S3PackageData`'s method should accept ecosystem package parameters. The logic should be encapsulated so callee doesn't need to know internal structure of how data are stored - so no prefix argument should be passed.

Reply:
Ok, i see what you mean. Done the necessary changes. Let me know if you are ok